lib/orocos/process.rb

```diff
@@ -220,26 +230,40 @@ def each_task
 
         # Returns the TaskContext instance for a task that runs in this process,
         # or raises Orocos::NotFound.
-        def task(task_name, name_service = Orocos.name_service)
+        def task(task_name, name_service = nil)
             full_name = "#{name}_#{task_name}"
-            if result = tasks.find { |t| t.basename == task_name || t.basename == full_name }
+            if (result = tasks.find { |t| [task_name, full_name].include?(t.basename) })
                 return result
             end
 
-            if task_names.include?(task_name)
-                name_service.get task_name, process: self
-            elsif task_names.include?(full_name)
-                name_service.get full_name, process: self
-            else
-                raise Orocos::NotFound, "no task #{task_name} defined on #{name}"
+            if name_service
+                Orocos.warn(
+                    "The name_service argument is deprecated and it's being ignored."
+                )
+            end
+
+            ior = ior_for(task_name) || ior_for(full_name)
+            unless ior
+                raise Orocos::IORNotRegisteredError,
+                      "no IOR is registered for #{task_name}"
             end
+
+            Orocos::TaskContext.new(ior, process: self)
         end
 
         def register_task(task)
             @tasks.delete_if { |t| t.name == task.name }
             @tasks << task
         end
 
+        def register_ior(task_name, ior)
+            @ior_mappings[task_name] = ior
+        end
```

Comment:
If we use `@ior_mappings == nil` as a way to say that the deployment is not ready, you need to set it all or not at all, can't set it task-by-task